### DISCUSSION TITLE
Additional Error capturing from connectors

### DISCUSSION THREAD

**Discussion by cb-alfredjoseph** [juspay/hyperswitch#4094] [Ideas / Feature Requests]
Currently we are collecting only two fields(error_code, error_message) for capturing error details from connector, Payment Gateways provide more details about declines such as network errors, processor errors and error params.

For Example : Nuvei is sending below fields

gwErrorCode 
gwErrorReason
gwExtendedErrorCode 
issuerDeclineCode
issuerDeclineReason

But currently  at HS we are capturing only gwErrorCode and gwErrorReason, we need more fields to capture the rest of the errors.

sample [docs](https://apidocs.chargebee.com/docs/api/transactions?lang=curl#transaction_error_detail) , [References](https://apidocs.chargebee.com/docs/api/gateway_error_references)

**Comment by ArjunKarthik** [comment]
Hi @cb-alfredjoseph, Thanks for initiating a discussion over here. We are evaluating this and will get back to you as soon as possible.
